### ISSUE TITLE
Static variables in Stream_support package 

### ISSUE THREAD

**Issue by sloriot** [CGAL/cgal#1406]
- [x] `Stream_support/include/CGAL/IO/io.h:46:  CGAL_EXPORT static int mode;`
- [x] `Stream_support/include/CGAL/IO/io.h:52:    static int mode = std::ios::xalloc();`

**Comment by afabri** [comment]
The mode should be  `static CGAL::cpp11::atomic` but I have no idea how to initialize it. @lrineau can you please help me.

**Comment by lrineau** [comment]
It certainly should not be atomic! That variable is supposed to be a constant, dynamically allocated using `std::ios::xalloc`. Make it `static const`, so that it is initialized before `main()` is called. See again:
http://en.cppreference.com/w/cpp/language/initialization#Non-local_variables
> All non-local variables with static storage duration are initialized as part of program startup, before the execution of the main function begins (unless deferred, see below).

Well, there is an *implementation-defined* possibility that static variables are initialized at first use. That is what is hidden in the [*unless deferred, see below*][*]. Let's assume that variable will be actually be initialized before the first thread starts.

[*]: http://en.cppreference.com/w/cpp/language/initialization#Non-local_variables

And `get_static_mode` should return an `int` instead of a non-const reference to an `int`.

If it happens there is an issue with the initialization of the variable returned by `get_static_mode()`, we'll make that variable an `atomic<int>`, whose initialization would be protected by a mutex. But I think that would be too much, for the moment.

**Comment by afabri** [comment]
It is a non-const reference as it serves as getter and setter at the same time, because inside is the static variable.

**Comment by lrineau** [comment]
The only uses of `get_static_mode` are:
```C++
i.iword(IO::get_static_mode())
```
That is a getter. The only place were that variable is set is its initialization.
```C++
static int mode = std::ios::xalloc();
```

**Comment by lrineau** [comment]
Fixed by #1651.
